articles/hashtables-implementation-using-CRC32-algorithm/index.md

```diff
@@ -0,0 +1,363 @@
+Hash tables are the most and convenient data structures that are easily accessible in most programming languages. It uses an associative array to store data and retrieve data. Each data value has its unique index. Instead of searching through the whole array, the index is used to access the required element directly. Time taken for search operation is reduced irrespective of the data size. Given a key-value pair, the hash code of the key is computed and used as the index of the value stored in the table.
+
+### Prerequisites
+
+To get started, you will need to understand Python arrays, lists, and classes. You also need a code editor and a Python interpreter to run your code. We will be using visual studio code, which has an integrated terminal for running the code. You can also use PyCharm for the same.
+
+### 1. The HashTable class members
+
+First, we will need to have all the Python modules that we will use in our environment. We will import `randint`,`Typevar`, `Callable`, and `List`.
+
+```python
+from random import randint
+from typing import TypeVar, Callable,  List
+```
+
+We then define our `HashTable` class with its data members and methods.
+
+```python
+class HashTable:
+    def __init__(self):
+        # Initial table size
+        self.table_size: int = 23
+
+        # initilizing all table slots to none
+        self.table: List[(T, T)] = [None] * self.table_size
```

Comment:
The variable T is not defined. Can you check the entire code once? I am getting errors
1. Indentation errors
2. Name Errors. 

Add a repl link which will ensure that we know the code is working. For example:

Run this code [here](https://repl.it/@lalithNarayan/SuperiorDeficientType).